core/lib/multivm/src/versions/vm_latest/utils/overhead.rs

```diff
@@ -1,356 +1,9 @@
-    // OB * TL + EP * MP > OE * EP * MP + OE * OB
-    // (OB * TL + EP * MP) / (EP * MP + OB) > OE
-    // OE = floor((OB * TL + EP * MP) / (EP * MP + OB)) with possible -1 if the division is without remainder
-    // let overhead_for_pubdata = {
-    //     let numerator: U256 = overhead_for_block_gas * total_gas_limit
-    //         + gas_per_pubdata_byte_limit * U256::from(MAX_PUBDATA_PER_BLOCK);
-    //     let denominator =
-    //         gas_per_pubdata_byte_limit * U256::from(MAX_PUBDATA_PER_BLOCK) + overhead_for_block_gas;
-    //
-    //     // Corner case: if `total_gas_limit` = `gas_per_pubdata_byte_limit` = 0
-    //     // then the numerator will be 0 and subtracting 1 will cause a panic, so we just return a zero.
-    //     if numerator.is_zero() {
-    //         0.into()
-    //     } else {
-    //         (numerator - 1) / denominator
-    //     }
-    // };
-    //
-    // 4. K * ceil(O4 * overhead_for_block_gas) >= overhead_gas, where K is the discount
-    // O4 = gas_limit / MAX_TX_ERGS_LIMIT. Using the notation from the previous equation:
-    // ceil(OB * GL / MAX_TX_ERGS_LIMIT) >= (OE / K)
-    // ceil(OB * (TL - OE) / MAX_TX_ERGS_LIMIT) >= (OE/K)
-    // OB * (TL - OE) / MAX_TX_ERGS_LIMIT > (OE/K) - 1
-    // OB * (TL - OE) > (OE/K) * MAX_TX_ERGS_LIMIT - MAX_TX_ERGS_LIMIT
-    // OB * TL + MAX_TX_ERGS_LIMIT > OE * ( MAX_TX_ERGS_LIMIT/K + OB)
-    // OE = floor(OB * TL + MAX_TX_ERGS_LIMIT / (MAX_TX_ERGS_LIMIT/K + OB)), with possible -1 if the division is without remainder
-    // ```
-    let overhead_for_gas = {
-        let numerator = overhead_for_block_gas * total_gas_limit + U256::from(MAX_TX_ERGS_LIMIT);
-        let denominator: U256 = U256::from(
-            (MAX_TX_ERGS_LIMIT as f64 / coefficients.ergs_limit_overhead_coeficient) as u64,
-        ) + overhead_for_block_gas;
-
-        let overhead_for_gas = (numerator - 1) / denominator;
-
-        overhead_for_gas.as_u32()
-    };
-
-    let overhead = vec![tx_slot_overhead, overhead_for_length, overhead_for_gas]
-        .into_iter()
-        .max()
-        // For the sake of consistency making sure that total_gas_limit >= max_overhead
-        .map(|max_overhead| std::cmp::min(max_overhead, total_gas_limit.as_u32()))
-        .unwrap();
-
-    let limit_after_deducting_overhead = total_gas_limit - overhead;
-
-    // During double checking of the overhead, the bootloader will assume that the
-    // body of the transaction does not have any more than `MAX_L2_TX_GAS_LIMIT` ergs available to it.
-    if limit_after_deducting_overhead.as_u64() > MAX_L2_TX_GAS_LIMIT {
-        // We derive the same overhead that would exist for the `MAX_L2_TX_GAS_LIMIT` ergs
-        derive_overhead(
-            MAX_L2_TX_GAS_LIMIT as u32,
-            gas_per_pubdata_byte_limit,
-            encoded_len.as_usize(),
-            coefficients,
-        )
-    } else {
-        overhead
-    }
+pub(crate) fn derive_overhead(encoded_len: usize) -> u32 {
```

Comment:
Add a comment.

And mention that in the past we used different rules, but with the new fee model, we only consider encoded len - as other part (like slots, memory, gas and pubdata are computed somewhere else)